haystack/components/routers/metadata_router.py

```diff
@@ -2,14 +2,17 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-from typing import Dict, List
+from typing import Any, Dict, Generic, List, Type, TypeVar, cast
 
 from haystack import Document, component
+from haystack.dataclasses.byte_stream import ByteStream
 from haystack.utils.filters import document_matches_filter
 
+T = TypeVar("T", Document, ByteStream)
+
 
 @component
-class MetadataRouter:
+class MetadataRouter(Generic[T]):
```

Comment:
Why are we inheriting from `Generic` here? I don't think this is needed. 
```suggestion
class MetadataRouter:
```